### PR TITLE
Use native Object.assign()

### PR DESCRIPTION
**Pull request by arusakov**
* remove `object-assign` dependence
* add engines.node >= 4.0 to `package.json`
* add node 7 for travis
* switch TS target to es5 (if we use node >=4.0 it's good, in future es6 target for node >= 6.0)

### CODE REVIEW

**Comment by johnnyreilly** [comment]
Hi @arusakov,

Thanks for the contribution - it's appreciated.   That said, I don't see the advantage of taking this PR at the moment I'm afraid.  Whilst node 0.12 is being end-of-lifed I'm aware there's people out there who will take a little longer to get to the point of upgrade and so I'll probably wait before making this move.  I hope you understand.

For what it's worth, `object-assign` hands over to native `Object.assign` when it's available.

**Comment by arusakov** [comment]
HI @johnnyreilly,
Thank you for answer. From my point of view `node 0.12` is a history. I think that we need go ahead. Maybe this breaking changes for `ts-loader 2.0`?

**Comment by johnnyreilly** [comment]
> From my point of view `node 0.12` is a history. 

That's fine - but you don't speak for everyone :smile:

**Comment by johnnyreilly** [comment]
Given webpack 2 is dropping support for node 0.12.0 we'll probably look at merging this soon.  One question: why "lib": "es2015"?  For Object.assign itself?

**Comment by arusakov** [comment]
@johnnyreilly yes, for definition of Object.assign

**Comment by johnnyreilly** [comment]
Cool - thanks. Will bump ts-loader to 2.0 as part of the merge. (Probably in a couple of weeks time)

File: src/tsconfig.json [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+        "target": "es5",
         "noImplicitAny": true,
         "noImplicitReturns": true,
         "noImplicitThis": false,
         "noUnusedLocals": true,
         "noUnusedParameters": true,
         "suppressImplicitAnyIndexErrors": true,
         "strictNullChecks": false,
+        "lib": [
+            "dom", "es2015" // dom for console.log only
```
Comment (HerringtonDarkholme) [inline src/tsconfig.json]: We probably can use `es2015.core`.
Also, we can shun away from "dom" like here https://github.com/Microsoft/TypeScript/pull/13453/files

Comment (johnnyreilly) [inline src/tsconfig.json]: so do you think this could be replaced with:
` "lib": [
  "es2015.core", "scripthost"
]`
?

Comment (johnnyreilly) [inline src/tsconfig.json]: I'd certainly rather use that than `dom` if that's possible

Comment (HerringtonDarkholme) [inline src/tsconfig.json]: Probably `["es5", "es2015.core"]` with this declaration 

`declare var console: {log(...args: any[]): void}`

Comment (johnnyreilly) [inline src/tsconfig.json]: I'll probably merge and then do some tweaking

Comment (arusakov) [inline src/tsconfig.json]: @johnnyreilly I will rebase and fix it today

**Pull request merged**